cmd/abapLandscapePortalUpdateAddOnProduct.go

```diff
@@ -0,0 +1,453 @@
+}
+
+type reqEntity struct {
+	RequestId string `json:"RequestId"`
+	ZoneId    string `json:"zone_id"`
+	Status    string `json:"Status"`
+	SystemId  string `json:"SystemId"`
+}
+
+type updateAddOnReq struct {
+	ProductName    string `json:"productName"`
+	ProductVersion string `json:"productVersion"`
+}
+
+type updateAddOnResp struct {
+	RequestId string `json:"requestId"`
+	ZoneId    string `json:"zoneId"`
+	Status    string `json:"status"`
+	SystemId  string `json:"systemId"`
+}
+
+func abapLandscapePortalUpdateAddOnProduct(config abapLandscapePortalUpdateAddOnProductOptions, telemetryData *telemetry.CustomData) {
+	client := http.Client{}
+
+	// Error situations should be bubbled up until they reach the line below which will then stop execution
+	// through the log.Entry().Fatal() call leading to an os.Exit(1) in the end.
+	err := runAbapLandscapePortalUpdateAddOnProduct(&config, client)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapLandscapePortalUpdateAddOnProduct(config *abapLandscapePortalUpdateAddOnProductOptions, client http.Client) error {
+	var systemId, reqId, reqStatus string
+	var clientAT http.Client
+	var servKey serviceKey
+	var getStatusReq http.Request
+	var err error
+
+	if prepareErr := parseServiceKeyAndPrepareAccessTokenHttpClient(config.LandscapePortalAPIServiceKey, &clientAT, &servKey); prepareErr != nil {
+		err = fmt.Errorf("Failed to prepare credentials to get access token of LP API. Error: %v\n", prepareErr)
+		return err
+	}
+
+	// get system
+	if getSystemErr := getSystemBySystemNumber(config, client, clientAT, servKey, &systemId); getSystemErr != nil {
+		err = fmt.Errorf("Failed to get system with systemNumber %v. Error: %v\n", config.AbapSystemNumber, getSystemErr)
+		return err
+	}
+
+	// update addon in the system
+	if updateAddOnErr := updateAddOn(config, client, clientAT, servKey, systemId, &reqId); updateAddOnErr != nil {
+		err = fmt.Errorf("Failed to update addon in the system with systemId %v. Error: %v\n", systemId, updateAddOnErr)
+		return err
+	}
+
+	// prepare http request to poll status of addon update
+	if prepareGetStatusHttpRequestErr := prepareGetStatusHttpRequest(clientAT, servKey, reqId, &getStatusReq); prepareGetStatusHttpRequestErr != nil {
+		err = fmt.Errorf("Failed to prepare http request to poll status of addon update request %v. Error: %v\n", reqId, prepareGetStatusHttpRequestErr)
+		return err
+	}
+
+	// query status of request
+	if pollStatusOfUpdateAddOnErr := pollStatusOfUpdateAddOn(client, &getStatusReq, reqId, &reqStatus); pollStatusOfUpdateAddOnErr != nil {
+		err = fmt.Errorf("Failed to poll status of addon update request %v. Error: %v\n", reqId, pollStatusOfUpdateAddOnErr)
+		return err
+	}
+
+	// keep polling status of addon update request until it reaches a final status (C/E/X)
+	for reqStatus == StatusInProgress || reqStatus == StatusScheduled {
+		// poll status every 30s
+		time.Sleep(30 * time.Second)
```

Comment:
30 is a magic number in terms of clean code and should be avoided - practically it slows down every unit test which is executing this function dramatically for no obvious reason. I would suggest to either define a configuration parameter for the polling interval or define the polling interval in function abapLandscapePortalUpdateAddOnProduct and provide it as variable of type time.Duration into the runAbapLandscapePortalUpdateAddOnProduct. The runAbapLandscapePortalUpdateAddOnProduct function should be tested by unit tests. So defining the polling interval outside lets you set it to time.Duration(1)*time.Millisecond so it takes no time in unit tests.